### ISSUE TITLE
bad training with monosdf

### ISSUE THREAD

**Issue by Juanjlp1109** [autonomousvision/sdfstudio#174]
**Describe the bug**
the training was a bad evaluation in the wandb visor. It seems that the training was make in a bad way.

**To Reproduce**
1. First process the data with Colmap in Nerfstudio like this:
ns-process-data images --data /home/lidarit/Desktop/Oficina_SamsugS21 --output-dir /home/lidarit/Desktop/Salidad_Oficina
IMPORTANT: Colmap just can make the match of the almost 70% of the images

2. obtained depth and normal maps with omnidata:
python scripts/datasets/process_nerfstudio_to_sdfstudio.py --data Oficina_SamsugS21/Salidad_Oficina --output-dir Oficina_SamsugS21/Salidad_Oficina/omnidata --scene-type indoor --data-type colmap --mono-prior --crop-mult 1 --omnidata-path omnidata/omnidata_tools/torch --pretrained-models omnidata/omnidata_tools/torch/pretrained_models

3. ns-train monosdf --pipeline.model.sdf-field.use-grid-feature True --pipeline.model.sdf-field.hidden-dim 256 --pipeline.model.sdf-field.num-layers 2 --pipeline.model.sdf-field.num-layers-color 2 --pipeline.model.sdf-field.use-appearance-embedding True --pipeline.model.sdf-field.geometric-init True --pipeline.model.sdf-field.inside-outside True  --pipeline.model.sdf-field.bias 0.8 --pipeline.model.sdf-field.beta-init 0.1 --pipeline.datamanager.train-num-images-to-sample-from 1 --pipeline.datamanager.train-num-times-to-repeat-images 0 --trainer.steps-per-eval-image 5000 --pipeline.model.background-model none --vis wandb --experiment-name Prueba_samsungS21 --pipeline.model.mono-depth-loss-mult 0.1 --pipeline.model.mono-normal-loss-mult 0.05 --pipeline.datamanager.train-num-rays-per-batch 2048 --machine.num-gpus 1 sdfstudio-data --data Oficina_SamsugS21/Salidad_Oficina/omnidata --include_mono_prior True --skip_every_for_val_split 30

**Expected behavior**
I had made other trainers with the same parameters and I obteined good results

**Screenshots**
these are the evaluations
![image](https://github.com/autonomousvision/sdfstudio/assets/136670910/b6edb7af-13a9-4d60-a0d4-b45941da7152)
![image](https://github.com/autonomousvision/sdfstudio/assets/136670910/cb9f0275-1a2d-4417-85ff-67cad3a3efc0)

**Additional context**

**Comment by niujinshuchong** [comment]
Hi, the normal and depth prior looks good. Could yo try to disable feature grids by `--pipeline.model.sdf-field.use-grid-feature False` as using of feature grids is sensitive to different hyper parameters. And also test only using normal priors and disable depth priors.

**Comment by Juanjlp1109** [comment]
disable depth priors is set pipeline....depth loss = 0 ?

**Comment by niujinshuchong** [comment]
Yes, `--pipeline.model.mono-depth-loss-mult 0.0`.

**Comment by Juanjlp1109** [comment]
I tried with depth-loss=0 and --pipeline.model.sdf-field.use-grid-feature False and got this results
I also aggregate this 2 parameters, idk if this combination may generate confusion in the model.
ns-train monosdf --pipeline.model.sdf-field.use-grid-feature False --pipeline.model.sdf-field.hidden-dim 256 --pipeline.model.sdf-field.num-layers 2 --pipeline.model.sdf-field.num-layers-color 2 --pipeline.model.sdf-field.use-appearance-embedding True --pipeline.model.sdf-field.geometric-init True --pipeline.model.sdf-field.inside-outside True  --pipeline.model.sdf-field.bias 0.8 --pipeline.model.sdf-field.beta-init 0.1 --pipeline.datamanager.train-num-images-to-sample-from 1 --pipeline.datamanager.train-num-times-to-repeat-images 0 --trainer.steps-per-eval-image 5000 --pipeline.model.background-model none --vis wandb --experiment-name The_office --pipeline.model.mono-normal-loss-mult 0.05 --pipeline.datamanager.train-num-rays-per-batch 2048 --pipeline.model.eikonal-loss-mult 0.01 --machine.num-gpus 1 sdfstudio-data --data The_office/omnidata --include_mono_prior True --skip_every_for_val_split 30
This are some of the results
/home/lidarit/Pictures/Screenshot from 2023-07-25 14-56-44.png
/home/lidarit/Pictures/Screenshot from 2023-07-25 14-57-54.png
This is the mesh texturized.
Low quality in monosdf training process

**Comment by niujinshuchong** [comment]
Hi, now the results looks better than the first one you shared? Maybe you could also use a large network `--pipeline.model.sdf-field.num-layers 2` when disable feature grids.

**Comment by Juanjlp1109** [comment]
I already used --pipeline.model.sdf-field.num-layers 2, can I improve this?
Other question, how can I work with 2 gpus? what information I have to provide about the second gpu or what configuration I have to do?

**Comment by niujinshuchong** [comment]
Oh sorry, I mean `--pipeline.model.sdf-field.num-layers 8`.

Multi-gpu can be easily used with `--machine.num-gpus 2`.

**Comment by Juanjlp1109** [comment]
ok, I will try with the large network
But, how I connect the other pc, I mean the other gpu, I don't need give the number IP or something?
I want to know what documentation I have to learn for understand better the parameters? Thanks

**Comment by Juanjlp1109** [comment]
I mean for example, why you say that is better to use a large network when I have feature networks disable, and why is a possible solution for my problem, I'll appreciate that you can guide with that

**Comment by niujinshuchong** [comment]
Hi, you can use `ns-train monosdf --help` to check all the configurable parameters and all of them come with a short explanation. 

The current code base might supports multiple machines but unfortunately I only tested with single machine with multiple gpus.

**Comment by niujinshuchong** [comment]
A large network is more expressive and therefore can represent more details. Based on my own experiments, using hash feature grids is sensitive/not robust to initialization/hyper-parameters and using only a MLP with positional encoding is more robust.

**Comment by Juanjlp1109** [comment]
Maybe can I use accelerate config to use multiple gpus like in zip nerf?

**Comment by niujinshuchong** [comment]
I don't think accelerate is supported.

**Comment by niujinshuchong** [comment]
But if you have multi gpus in a single machine, that's already supported and you simply use `--machine.num-gpus 2`.

**Comment by lxr319** [comment]
> **Describe the bug** the training was a bad evaluation in the wandb visor. It seems that the training was make in a bad way.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. First process the data with Colmap in Nerfstudio like this:
>    ns-process-data images --data /home/lidarit/Desktop/Oficina_SamsugS21 --output-dir /home/lidarit/Desktop/Salidad_Oficina
>    IMPORTANT: Colmap just can make the match of the almost 70% of the images
> 2. obtained depth and normal maps with omnidata:
>    python scripts/datasets/process_nerfstudio_to_sdfstudio.py --data Oficina_SamsugS21/Salidad_Oficina --output-dir Oficina_SamsugS21/Salidad_Oficina/omnidata --scene-type indoor --data-type colmap --mono-prior --crop-mult 1 --omnidata-path omnidata/omnidata_tools/torch --pretrained-models omnidata/omnidata_tools/torch/pretrained_models
> 3. ns-train monosdf --pipeline.model.sdf-field.use-grid-feature True --pipeline.model.sdf-field.hidden-dim 256 --pipeline.model.sdf-field.num-layers 2 --pipeline.model.sdf-field.num-layers-color 2 --pipeline.model.sdf-field.use-appearance-embedding True --pipeline.model.sdf-field.geometric-init True --pipeline.model.sdf-field.inside-outside True  --pipeline.model.sdf-field.bias 0.8 --pipeline.model.sdf-field.beta-init 0.1 --pipeline.datamanager.train-num-images-to-sample-from 1 --pipeline.datamanager.train-num-times-to-repeat-images 0 --trainer.steps-per-eval-image 5000 --pipeline.model.background-model none --vis wandb --experiment-name Prueba_samsungS21 --pipeline.model.mono-depth-loss-mult 0.1 --pipeline.model.mono-normal-loss-mult 0.05 --pipeline.datamanager.train-num-rays-per-batch 2048 --machine.num-gpus 1 sdfstudio-data --data Oficina_SamsugS21/Salidad_Oficina/omnidata --include_mono_prior True --skip_every_for_val_split 30
> 
> **Expected behavior** I had made other trainers with the same parameters and I obteined good results
> 
> **Screenshots** these are the evaluations ![image](https://user-images.githubusercontent.com/136670910/255293892-b6edb7af-13a9-4d60-a0d4-b45941da7152.png) ![image](https://user-images.githubusercontent.com/136670910/255293951-cb9f0275-1a2d-4417-85ff-67cad3a3efc0.png)
> 
> **Additional context**

Hello, first of all thank you for sharing, I according to your steps down, training the camera pose is that how to deal with this?
![image](https://github.com/autonomousvision/sdfstudio/assets/45560167/814866ed-310b-4632-a60e-4bf69d0f3b93)

**Comment by niujinshuchong** [comment]
@lxr319 You can enable auto-orient by appending `--auto-orient True` to your training command.
